build(weaver/core/relay): fix container image build due to OpenSSL version

### Description

The OpenSSL version that the container images are hardcoded to use via URL downloading has been moved to a different link and the container image builds fail due to this.

The old versions that the images are trying to use are now under this URL path:
https://www.openssl.org/source/old/1.1.1/index.html

### Logs

```sh
#20 [builder 8/9] RUN mkdir /musl &&     curl https://www.openssl.org/source/openssl-1.1.1f.tar.gz -O &&     tar zxvf openssl-1.1.1f.tar.gz &&     cd openssl-1.1.1f/ &&     CC="musl-gcc -fPIE -pie" ./Configure no-shared no-async --prefix=/musl --openssldir=/musl/ssl linux-x86_64 &&     make depend && make -j$(nproc) && make install
#20 0.195   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#20 0.195                                  Dload  Upload   Total   Spent    Left  Speed
#20 0.195 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  4853  100  4853    0     0  19568      0 --:--:-- --:--:-- --:--:-- 19568
#20 0.448 
#20 0.448 gzip: stdin: not in gzip format
#20 0.448 tar: Child returned status 1
#20 0.448 tar: Error is not recoverable: exiting now
#20 ERROR: process "/bin/sh -c mkdir /musl &&     curl https://www.openssl.org/source/openssl-$SSL_VERSION.tar.gz -O &&     tar zxvf openssl-$SSL_VERSION.tar.gz &&     cd openssl-$SSL_VERSION/ &&     CC=\"musl-gcc -fPIE -pie\" ./Configure no-shared no-async --prefix=/musl --openssldir=/musl/ssl linux-x86_64 &&     make depend && make -j$(nproc) && make install" did not complete successfully: exit code: 2
------
 > [builder 8/9] RUN mkdir /musl &&     curl https://www.openssl.org/source/openssl-1.1.1f.tar.gz -O &&     tar zxvf openssl-1.1.1f.tar.gz &&     cd openssl-1.1.1f/ &&     CC="musl-gcc -fPIE -pie" ./Configure no-shared no-async --prefix=/musl --openssldir=/musl/ssl linux-x86_64 &&     make depend && make -j$(nproc) && make install:

100  4853  100  4853    0     0  19568      0 --:--:-- --:--:-- --:--:-- 19568
0.448 
0.448 gzip: stdin: not in gzip format
0.448 tar: Child returned status 1
0.448 tar: Error is not recoverable: exiting now
------
Dockerfile.server:26
--------------------
  25 |         ln -s /usr/include/linux /usr/include/x86_64-linux-musl/linux
  26 | >>> RUN mkdir /musl && \
  27 | >>>     curl https://www.openssl.org/source/openssl-$SSL_VERSION.tar.gz -O && \
  28 | >>>     tar zxvf openssl-$SSL_VERSION.tar.gz && \
  29 | >>>     cd openssl-$SSL_VERSION/ && \
  30 | >>>     CC="musl-gcc -fPIE -pie" ./Configure no-shared no-async --prefix=/musl --openssldir=/musl/ssl linux-x86_64 && \
  31 | >>>     make depend && make -j$(nproc) && make install
  32 |         
--------------------
ERROR: failed to solve: process "/bin/sh -c mkdir /musl &&     curl https://www.openssl.org/source/openssl-$SSL_VERSION.tar.gz -O &&     tar zxvf openssl-$SSL_VERSION.tar.gz &&     cd openssl-$SSL_VERSION/ &&     CC=\"musl-gcc -fPIE -pie\" ./Configure no-shared no-async --prefix=/musl --openssldir=/musl/ssl linux-x86_64 &&     make depend && make -j$(nproc) && make install" did not complete successfully: exit code: 2
make: *** [Makefile:90: build-server-local] Error 1
```

### More Context

> Note: The latest stable version is the 3.2 series supported until 23rd November 2025. Also [available](https://www.openssl.org/support/contracts.html) is the 3.1 series supported until 14th March 2025, and the 3.0 series which is a Long Term Support (LTS) version and is supported until 7th September 2026. All older versions (including 1.1.1, 1.1.0, 1.0.2, 1.0.0 and 0.9.8) are now out of support and should not be used. Users of these older versions are encouraged to upgrade to 3.2 or 3.0 as soon as possible. Extended support for 1.1.1 and 1.0.2 to gain access to security fixes for those versions is available.

Source: https://www.openssl.org/source/


### Acceptance Criteria
1. Container image build works

--- a/weaver/core/relay/Dockerfile
+++ b/weaver/core/relay/Dockerfile
@@ -20,8 +20,8 @@ COPY . /opt/relay
 
 # Build OpenSSL
 
-ENV SSL_VERSION=1.1.1f
-    
+ENV SSL_VERSION=1.1.1w
+    # https://www.openssl.org/source/openssl-1.1.1w.tar.gz
 RUN ln -s /usr/include/x86_64-linux-gnu/asm /usr/include/x86_64-linux-musl/asm && \
     ln -s /usr/include/asm-generic /usr/include/x86_64-linux-musl/asm-generic && \
     ln -s /usr/include/linux /usr/include/x86_64-linux-musl/linux

--- a/weaver/core/relay/Dockerfile.client
+++ b/weaver/core/relay/Dockerfile.client
@@ -20,7 +20,7 @@ COPY . /opt/relay
 
 # Build OpenSSL
 
-ENV SSL_VERSION=1.1.1f
+ENV SSL_VERSION=1.1.1w
     
 RUN ln -s /usr/include/x86_64-linux-gnu/asm /usr/include/x86_64-linux-musl/asm && \
     ln -s /usr/include/asm-generic /usr/include/x86_64-linux-musl/asm-generic && \

--- a/weaver/core/relay/Dockerfile.driver
+++ b/weaver/core/relay/Dockerfile.driver
@@ -20,7 +20,7 @@ COPY . /opt/relay
 
 # Build OpenSSL
 
-ENV SSL_VERSION=1.1.1f
+ENV SSL_VERSION=1.1.1w
     
 RUN ln -s /usr/include/x86_64-linux-gnu/asm /usr/include/x86_64-linux-musl/asm && \
     ln -s /usr/include/asm-generic /usr/include/x86_64-linux-musl/asm-generic && \

--- a/weaver/core/relay/Dockerfile.server
+++ b/weaver/core/relay/Dockerfile.server
@@ -18,7 +18,7 @@ COPY . /opt/relay
 
 # Build OpenSSL
 
-ENV SSL_VERSION=1.1.1f
+ENV SSL_VERSION=1.1.1w
     
 RUN ln -s /usr/include/x86_64-linux-gnu/asm /usr/include/x86_64-linux-musl/asm && \
     ln -s /usr/include/asm-generic /usr/include/x86_64-linux-musl/asm-generic && \

